operator/internal/controller/pulumi/stack_controller.go

```diff
@@ -974,11 +978,11 @@ func (r *StackReconciler) markStackSucceeded(ctx context.Context, instance *pulu
 				return fmt.Errorf("unmarshaling output mask: %w", err)
 			}
 		}
-		for key, value := range secret.Data {
+		for _, key := range slices.Sorted(maps.Keys(secret.Data)) {
```

Comment:
Do we really need to sort here? We're ultimately updating another map based on the keys of another map, so order does not change the output.